### DISCUSSION TITLE
Could rendered copies of older Superset versions' docs be made available on https://superset.apache.org?

### DISCUSSION THREAD

**Discussion by vergenzt** [apache/superset#27604] [Ideas]
I was trying to open the doc linked in the (fairly recent) discussion at https://github.com/apache/superset/discussions/26165#discussioncomment-7757833, but encountered a 404 because the doc was removed in the latest Superset version. The doc was about how to migrate away from a deprecated component (which was now _removed_ in the latest version) -- which seems like it'd be handy to have accessible. 🙂

While it is currently possible to still view older docs it's a bit of a pain to find them, _and_ it's a bit of a pain to view them without the rendering. (They don't even render in Github because the extensions are `.mdx`! 😐)

Could older specific versions of the docs be rendered under something like https://superset.apache.org/docs/{major_version}/{page_slug}? As opposed to the latest version which is currently rendered under https://superset.apache.org/docs/{page_slug}.

E.g. https://superset.apache.org/docs/miscellaneous/native-filter-migration/ could be made available in rendered form at https://superset.apache.org/docs/v3/miscellaneous/native-filter-migration/

To be a bit more clean we could also do one or more of the following:

 1) Add a "latest" alias to whatever the current major version is (i.e. "v4" right now) so that https://superset.apache.org/docs/latest/miscellaneous/intro/ either:
    - a) _redirects_ to https://superset.apache.org/docs/v4/miscellaneous/intro/, or
    - b) renders the same as https://superset.apache.org/docs/v4/miscellaneous/intro/ without redirect
    - (I'm not sure which would be better)

 2) Add redirects from https://superset.apache.org/docs/{slug} to https://superset.apache.org/docs/latest/{slug}

**Comment by michael-s-molina** [comment]
@sfirke @rusackas

**Comment by vergenzt** [comment]
Note I'm happy to do some work toward contributing this -- I'd just want buy-in/approval and need info on logistical constraints from maintainers before doing so.

**Comment by sfirke** [comment]
Thanks for bringing this up, it was just being discussed elsewhere by @rusackas and others.  [Docusaurus supports versioned docs](https://docusaurus.io/docs/versioning).  However they start by warning against it:
> Most of the time, you don't need versioning as it will just increase your build time, and introduce complexity to your codebase. Versioning is best suited for websites with high-traffic and rapid changes to documentation between versions. If your documentation rarely changes, don't add versioning to your documentation.

I'm torn.  I don't want more complexity to manage.  At the same time, the current "solution" is not really working, which is: leave outdated material in the docs noting that it only applies to versions XXX-YYY.  In the case that you began with re: filter boxes, the page should not have been removed yet, as it was still relevant to people on older versions who may need to migrate away from filter boxes.  Instead we should have slapped "Relevant to <4.0.0" at the top ... but no one did that and it's hard to remember.

An example of this practice is at https://docusaurus.io/docs/versioning where the docs describe how to do something in 4.0.0 and in earlier versions.  That's kind of clunky, to be sure, and prone to getting stale.

Docusaurus notes that it's hard to make a change to older versions of the docs.  I don't see us ever doing that, honestly, I think we'd just snapshot the docs with each major release and never go back to update the old docs - and that would be okay.
